tiledb/sm/serialization/array_schema.cc

```diff
@@ -160,12 +188,24 @@ static tuple<Status, optional<shared_ptr<Filter>>> filter_constructor(
     case FilterType::FILTER_BZIP2:
     case FilterType::FILTER_DOUBLE_DELTA:
     case FilterType::FILTER_DICTIONARY: {
-      auto data = reader.getData();
+      auto data = filter_reader.getData();
       int32_t level = data.getInt32();
       return {
           Status::Ok(),
           tiledb::common::make_shared<CompressionFilter>(HERE(), type, level)};
     }
+    case FilterType::FILTER_SCALE_FLOAT: {
+      auto data = filter_reader.getData();
+      auto metadata_value = data.getBytes();
+      FloatScalingFilter::Metadata m;
+      memcpy(
```

Comment:
Adding the new capnp structure will force us remvoe the memcpy and use the usual capnp setters/getters